composition_test.go

```diff
@@ -242,7 +242,7 @@ func TestComposeBatchWontFail(t *testing.T) {
 
 	msg := <-c.Execute(makeMessage(StrMap{"input": 1}))
 	expected := StrMap{}
```

Comment:
@adityagodbole Receive is internal to gilmour whereas expected is what the caller-only knows what it should be unmarshalled into.
Same reason that json package cannot return any thing, it can only unmarshal the underlying interface to an instance of your choice, and returning error in case it fails to match the type.